src/averageAges.js

```diff
@@ -14,12 +14,37 @@
  *
  * @return {number}
  */
+
+// helper fn
+function isMan(person) {
+  return person.sex === 'm';
+}
+
+function sumAllAges(people) {
+  return people.reduce((sum, person) => sum + (person.died - person.born), 0);
+}
+
+function isWoman(person) {
+  return person.sex === 'f';
+};
+
+function isMother(person, index, arr) {
+  return arr.some(child => child.mother === person.name);
+};
+
+function haveDiedInCentury(person, century) {
+  return (century) ? (Math.ceil(person.died / 100)) === century : true;
+}
+
+function isChild(person, index, arr) {
+  return arr.some(child => child.name === person.mother);
+};
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people
+    .filter(person => isMan(person) && haveDiedInCentury(person, century));
```

Comment:
You could make those function accept only parts of the person object and destructure the person object in the callback params like so
```
.filter({sex, died} => isMan(sex) && haveDiedInCentury(died, century));
```